mktrayicon.c

```diff
@@ -256,10 +300,99 @@ gpointer watch_fifo(gpointer argv)
 #endif
 			break;
 		case 'm': /* menu */
-			fprintf(stderr, "Menu command not yet implemented\n");
-			if (param != NULL) {
-				free(param);
+			if (onmenu != NULL){
+				//destroy the previous menu
+				for (int i = 0; i < menusize; i++){
+					free(onmenu[i].name);
+					free(onmenu[i].action);
+					onmenu[i].name = NULL;
+					onmenu[i].action = NULL;
+				}
+				free(onmenu);
+				onmenu = NULL;
+				gtk_widget_destroy(menu);
+				menu = NULL;
 			}
+			
+			menusize = 0;
+
+			if (param != NULL && *param == '\0') {
+#ifdef DEBUG
+				printf("Removing onmenu handler\n");
+#endif
+				break;
+			}
+
+			// This block makes sure that the parameter after 'm' is well defined
+			int commas = 0;
+			int bars = 0;
+			if (len < 3){
+				break;
+			}
+			if (param[0] == ',' || param[0] == '|'){
+				printf("Action string cannot begin with ',' or '|'. Use '\\' to escape\n");
+				break;
+			}
+			if (param[len-2] == '|' && param[len-3] != '\\'){
+				printf("'|' can't be second of last. Use '\\' to escape\n");
```

Comment:
Is there any particular reason to explicitly disallow this? Won't it just be equivalent to having no defined command for a label? Even so, it's probably easier to detect this _after_ parsing than before.

Reply:
Not really. It just made the logic clearer for me. Because one character after a valid `|` at the end has no point (a label without action)

Reply:
I think labels without an action is actually a valid use-case. For example, if you want the context menu to show some additional information beyond what is shown in the tooltip.

Reply:
I checked and it works for all cases except the last entry (because of the explicit condition to disallow `,` at the end). It generally accepts `<label>,|` as a sequence, creating an empty entry doing nothing. I will see how the parsing loops work if I remove that condition.